### ISSUE TITLE
`Lint/UselessAssignment` false positive when variables are used in blocks with numbered parameters

### ISSUE THREAD

**Issue by Fustrate** [rubocop/rubocop#11485]
## Expected behavior

Variables that are set and then used inside blocks with numbered parameters should not be marked as unused.

It looks like it was caused by b48ebef423f74c3083bf2cd7ab79e8546f28a95e - I tried the commit before and it passed. This also causes `Lint/UnusedMethodArgument` and `Lint/UnusedBlockArgument` warnings.

## Actual behavior

```
For /Users/Fustrate/Development/DummyProject: configuration from /Users/Fustrate/Development/DummyProject/.rubocop.yml
configuration from /Users/Fustrate/.rbenv/versions/3.2.0/lib/ruby/gems/3.2.0/gems/rubocop-performance-1.15.2/config/default.yml
configuration from /Users/Fustrate/.rbenv/versions/3.2.0/lib/ruby/gems/3.2.0/gems/rubocop-performance-1.15.2/config/default.yml
Default configuration from /Users/Fustrate/.rbenv/versions/3.2.0/lib/ruby/gems/3.2.0/gems/rubocop-1.44.0/config/default.yml
configuration from /Users/Fustrate/.rbenv/versions/3.2.0/lib/ruby/gems/3.2.0/gems/rubocop-rails-2.17.4/config/default.yml
configuration from /Users/Fustrate/.rbenv/versions/3.2.0/lib/ruby/gems/3.2.0/gems/rubocop-rails-2.17.4/config/default.yml
Use parallel by default.
Skipping parallel inspection: only a single file needs inspection
Inspecting 1 file
Scanning /Users/Fustrate/Development/DummyProject/test.rb
W

Offenses:

test.rb:3:1: W: Lint/UselessAssignment: Useless assignment to variable - target.
target = 6
^^^^^^

1 file inspected, 1 offense detected
Finished in 0.47589999999945576 seconds
```

## Steps to reproduce the problem

```
# rubocop:disable all
# rubocop:enable Lint/UselessAssignment
target = 6

[1, 2, 3].each { puts _1 if _1 * 2 == target }

```

Changing it to a normal block with named arguments passes:

```
target = 6

[1, 2, 3].each do |num|
  puts  num if num * 2 == target
end
```

## RuboCop version

```
$ rubocop -V
1.44.0 (using Parser 3.2.0.0, rubocop-ast 1.24.1, running on ruby 3.2.0) [x86_64-darwin22]
  - rubocop-performance 1.15.2
  - rubocop-rails 2.17.4
```

Tested with both 1.44 and current master (99c8b1773df63a6fa7cce969f60a04026e0002ee)
